### PR TITLE
[v5] Rename `spawn(...)` action creator to `spawnChild(...)`

### PR DESCRIPTION
**Pull request by davidkpiano**
The `spawn(...)` action creator has been renamed to `spawnChild(...)` to avoid confusion.

```ts
import { spawnChild, assign } from 'xstate';

const childMachine = createMachine({
  on: {
    someEvent: {
      actions: [
        // spawnChild(...) instead of spawn(...)
        spawnChild('someSrc'),

        // spawn() is used inside of assign()
        assign({
          anotherRef: ({ spawn }) => spawn('anotherSrc')
        })
      ]
    }
  }
});
```

### CODE REVIEW

File: packages/core/src/actions/spawn.ts [hunk 1]
```diff
@@ -199,15 +199,15 @@ export function spawn<
     }
   }
 
-  spawn.type = 'xstate.spawn';
-  spawn.id = id;
-  spawn.systemId = systemId;
-  spawn.src = src;
-  spawn.input = input;
-  spawn.syncSnapshot = syncSnapshot;
+  spawnChild.type = 'xstate.spawn';
```
Comment (Andarist) [inline packages/core/src/actions/spawn.ts:202]:
```suggestion
  spawnChild.type = 'xstate.spawnChild';
```

File: packages/core/src/actions/spawn.ts
Comment (Andarist) [inline packages/core/src/actions/spawn.ts:1]: the filename could be renamed here as well

File: packages/core/src/serviceScope.ts [hunk 2]
```diff
@@ -1,22 +0,0 @@
-import { AnyActor } from './types.ts';
```
Comment (Andarist) [inline packages/core/src/serviceScope.ts:1]: oh wow, we still had this, lol

File: packages/core/src/serviceScope.ts
Comment (Andarist) [inline packages/core/src/serviceScope.ts:1]: oh wow, we still had this file lying around, lol

File: packages/core/test/spawn.test.ts [hunk 3]
```diff
@@ -6,14 +6,14 @@ import {
   fromObservable,
   fromPromise,
   sendTo,
-  spawn
+  spawnChild
 } from '../src';
 
 describe('spawn action', () => {
```
Comment (Andarist) [inline packages/core/test/spawn.test.ts:12]:
```suggestion
describe('spawnChild action', () => {
```

File: packages/core/test/types.test.ts
Comment (Andarist) [inline packages/core/test/types.test.ts:1]: describe block in this file should be retitled from `"spawn action"` to `"spawnChild action"`

File: packages/core/test/setup.types.test.ts
Comment (Andarist) [inline packages/core/test/setup.types.test.ts:1]: some test titles in this file still refer to `spawn` action, let's update them

File: yarn.lock
Comment (Andarist) [inline yarn.lock:1]: is `yarn.lock` broken on the `next` branch? 🤔 it's surprising that this PR updated it in any way

Comment (davidkpiano) [reply yarn.lock:1]: I had to update changesets

Comment (Andarist) [reply yarn.lock:1]: aaah, i see that you updated Changesets here, ok then

File: packages/core/src/actions/stopChild.ts [hunk 4]
```diff
@@ -92,7 +92,7 @@ export interface StopAction<
  *
  * @param actorRef The actor to stop.
  */
-export function stop<
+export function stopChild<
```
Comment (Andarist) [inline packages/core/src/actions/stopChild.ts:95]: we might want to export a deprecated `stop` alias from the `src/index.ts` for this

File: packages/core/src/spawnChild.ts
Comment (Andarist) [inline packages/core/src/spawnChild.ts:1]: u renamed the wrong `spawn` file 😅

File: packages/core/src/actions/stopChild.ts [hunk 5]
```diff
@@ -109,11 +109,20 @@ export function stop<
...
 
   return stop;
 }
+
+/**
+ * Stops a child actor.
+ *
+ * Deprecated: use `stopChild(...)` instead.
+ *
+ * @deprecated Use `stopChild(...)` instead
```
Comment (Andarist) [inline packages/core/src/actions/stopChild.ts:126]:
```suggestion
 * @deprecated Use `stopChild(...)` instead
```

**Review by Andarist — APPROVED** [review]

**Pull request merged**